### DISCUSSION TITLE
Build Failure: "npx react-native run-windows" -> ERROR: The system was unable to find the specified registry key or value.

### DISCUSSION THREAD

**Discussion by rustymiller** [microsoft/react-native-windows#11959] [Q&A] [answered]
I started from a pretty clean Win 11 machine (Microsoft Windows [Version 10.0.22621.1992]) and followed the setup instructions here: https://microsoft.github.io/react-native-windows/docs/getting-started .  The dependency check from https://microsoft.github.io/react-native-windows/docs/rnw-dependencies passed.

When I ran "npx react-native run-windows" it output:

```
ERROR: The system was unable to find the specified registry key or value.
...
Build failed with message 6:6>EXEC : error : The system was unable to find the specified registry key or value. [C:\Users\rusty\OneDrive\projects\src\client\MyApp\windows\MyApp\MyApp.vcxproj]. Check your build configuration.
```

Via trial and error I was able to address the first error by passing "--no-telemetry" -> "**npx react-native run-windows --no-telemetry**"
For the second error I needed to also update ...MyApp\node_modules\react-native-windows\PropertySheets\Autolink.props and ...MyApp\node_modules\react-native-windows\PropertySheets\Codegen.props to add "--no-telemetry" to the command args - for example:

```
<AutolinkCommandArgs Condition="'$(AutolinkCommandArgs)' == ''">--check --no-telemetry</AutolinkCommandArgs>
```

I can't rule out it's a specific issue to my machine causing the telemetry to fail - but wanted to share in case others are hitting the same issue.

**Comment by chrisglein** [comment]
Can you confirm which version of RNW you were running with this?

**Reply by rustymiller** [reply]
**npmPackages:
  react:**
    installed: 18.2.0
    wanted: 18.2.0
  **react-native:**
    installed: 0.72.3
    wanted: 0.72.3
  **react-native-windows:**
    installed: 0.72.4
    wanted: 0.72.4

**Reply by rustymiller** [reply]
I think it's due to my machine having 32bit node.js.  In node_modules\@react-native-windows\telemetry\lib-commonjs\utils\basePropUtils.js:

```
...
const DeviceIdRegPath = 'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\SQMClient';
const DeviceIdRegKey = 'MachineId';
/**
 * Gets a telemetry-safe stable device ID.
 * @returns A telemetry-safe stable device ID.
 */
async function deviceId() {
    try {
        const output = (0, child_process_1.execSync)(`${process.env.windir}\\System32\\reg.exe query ${DeviceIdRegPath} /v ${DeviceIdRegKey}`).toString();
        const result = output.match(/\{([0-9A-Fa-f-]{36})\}/);
        if (result && result.length > 1) {
            return `s:${result[1]}`;
        }
    }
    catch (_a) { }
    return '';
}
...
```

I can see from Process Monitor that it's looking for [HKLM\SOFTWARE\WOW6432Node\Microsoft\SQMClient] : "MachineId" (a redirected registry path since 32bit node.exe is launching 32bit reg.exe).  This key doesn't exist (at least on my machine).  However, I do have [HKLM\SOFTWARE\Microsoft\SQMClient] : "MachineId".

**Comment by jonthysell** [comment, marked as answer]
This is a bug, issue tracking it here: #11977
